diff --git a/rtl/ibex_core.sv b/rtl/ibex_core.sv
--- a/rtl/ibex_core.sv
+++ b/rtl/ibex_core.sv
@@ -110,7 +110,12 @@
     input  logic        fetch_enable_i,
     output logic        alert_minor_o,
     output logic        alert_major_o,
-    output logic        core_sleep_o
+    output logic        core_sleep_o,
+
+`ifdef CONTRACT
+    output logic        fetch_o,
+    output logic        retire_o,
+`endif
 );
 
   import ibex_pkg::*;
@@ -1471,4 +1476,11 @@
   // Certain parameter combinations are not supported
   `ASSERT_INIT(IllegalParamSecure, !(SecureIbex && (RV32M == RV32MNone)))
 
+`ifdef CONTRACT
+  logic granted = 0;
+  always @(posedge clk_i) granted <= instr_gnt_i;
+  assign fetch_o = granted;
+  assign retire_o = instr_id_done & ~dummy_instr_id;
+`endif
+
 endmodule
diff --git a/rtl/ibex_decoder.sv b/rtl/ibex_decoder.sv
--- a/rtl/ibex_decoder.sv
+++ b/rtl/ibex_decoder.sv
@@ -214,7 +214,7 @@
     csr_op                = CSR_OP_READ;
 
     data_we_o             = 1'b0;
-    data_type_o           = 2'b00;
+    data_type_o           = 2'b11;
     data_sign_extension_o = 1'b0;
     data_req_o            = 1'b0;
 
diff --git a/rtl/ibex_ex_block.sv b/rtl/ibex_ex_block.sv
--- a/rtl/ibex_ex_block.sv
+++ b/rtl/ibex_ex_block.sv
@@ -189,6 +189,10 @@
         .valid_o            ( multdiv_valid         ),
         .multdiv_result_o   ( multdiv_result        )
     );
+  end else begin : gen_multdiv_none
+    // Fix warnings about uninitialized wires
+    assign multdiv_alu_operand_a = 33'b0;
+    assign multdiv_alu_operand_b = 33'b0;
   end
 
   // Multiplier/divider may require multiple cycles. The ALU output is valid in the same cycle
